test/testutil.h

```diff
@@ -15,6 +15,7 @@
 #include <openssl/err.h>
 #include <openssl/e_os2.h>
 #include <openssl/bn.h>
+#include "../apps/opt.h"
```

Comment:
This turned out to be a bad idea, 'cause there are compilers out there that will fail on this, because they do *all* inclusions from the point of view of the C file being compiled.

So `test/testutil/basic_output.c` gets compiled, and it includes `"../testutil.h"`.  Totally fine so far, but then the compiler hits this line, and fails because it tries to find (seen from the source top) `test/apps/opt.h` instead of `apps/opt.h`.  #FAIL 

This leads me to think that we do need to restructure the header files associated with libapps.  I think @DDvO hinted at this quite some time ago, and it got dismissed at the time, but maybe it's time for a rethink that...

Reply:
What I proposed (originally) in May 2016 was something related but more fundamental, 
namely to have a proper application-oriented library on top of libssl: https://github.com/openssl/openssl/issues/6986
which could be (re-)used for the `openssl` apps, tests, and external application/libs.

Reply:
Hi Richard, Is there something you want me to do with this? test/Build.info includes could be updated to temporarily fix the issue. (Not that I could test that it worked from here).

Reply:
Yes, please have a look at #8210, which deals with this immediate problem